au4/src/projectscene/view/clipsview/au3/au3wavepainter.cpp

```diff
@@ -654,6 +659,14 @@ void Au3WavePainter::doPaint(QPainter& painter, const WaveTrack* _track, const W
     wm.fromTime = params.fromTime;
     wm.toTime = params.toTime;
 
+    auto selectedTracks = selectionController()->dataSelectedOnTracks();
+    bool isTrackSelected = std::find(selectedTracks.begin(), selectedTracks.end(), DomConverter::track(track).id) != selectedTracks.end();
+
+    if (isTrackSelected) {
+        wm.selectionStartTime = selectionController()->dataSelectedStartTime() - clip->Start() + clip->GetTrimLeft();
+        wm.selectionEndTime = selectionController()->dataSelectedEndTime() - clip->Start() + clip->GetTrimLeft();
+    }
```

Comment:
I think it would be good to add selectionStartTime and selectionEndTime to the parameters and pass them to Params from the outside (from ClipsListModel->ClipListItem->ClipTime) 

WaveView and Au3WavePainter - They should be treated as low-level components and have as few dependencies as possible. Just draw a wave with the given parameters...
Because they can be used in a variety of places, not just on the main clips screen.